### PR TITLE
Use "channel" instead of "branch" when referring to State Tool versions.

### PR DESCRIPTION
**Pull request by mitchell-as**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-207" title="DX-207" target="_blank">BugDX-207</a>  State tool consistent terminology (Branch VS Channel)
  </td></table>
  <br />
 

This tends to cause confusion because Platform projects have branches.

### CODE REVIEW

**Comment by mitchell-as** [comment]
Initial critical and installer tests passed. I've now kicked off "Test: all", but I'm confident this is ready for review.

**Comment by mitchell-as** [comment]
Failures are the usual timeouts and temporary Platform errors.

File: cmd/state/main.go [hunk 1]
```diff
@@ -242,9 +242,9 @@ func run(args []string, isInteractive bool, cfg *config.Instance, out output.Out
 
 		if childCmd.Name() != "update" && pj != nil && pj.IsLocked() {
 			if (pj.Version() != "" && pj.Version() != constants.Version) ||
-				(pj.VersionBranch() != "" && pj.VersionBranch() != constants.BranchName) {
+				(pj.VersionBranch() != "" && pj.VersionBranch() != constants.ChannelName) {
```
Comment (MDrakos) [inline cmd/state/main.go:245]: Should the project methods here also be renamed?

**Review by MDrakos — DISMISSED** [review]
Minor nit comment but otherwise looks fine to me.

File: cmd/state/main.go (hunk 1 above)
Comment (mitchell-as) [reply cmd/state/main.go:245]: No. You got caught by the purpose of this ticket: to eliminate confusion between channels (state tool version branches) and branches (Platform things). The project methods refer to Platform project branches, so they should not be changed.

Comment (mitchell-as) [reply cmd/state/main.go]: I was going to say no, but you're right. I thought all projectfile branch methods were related to Platform project branches, but apparently there's a way to lock projectfiles to State Tool channels. I learned something new.

**Review by MDrakos — APPROVED** [review]

**Pull request merged**